File: lib/layer2_core/light/e_secondary_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESecondaryLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl25;
      case 'standard': return cstl25;
      case 'blue': return cbl25;
      case 'green': return cgl25;
      case 'red': return crl25;
      case 'nothing': return cnol25;
      default: return cstl25;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESecondaryLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESecondaryLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl25
            AppTheme.STANDARD -> R.color.cstl25
            AppTheme.BLUE -> R.color.cbl25
            AppTheme.GREEN -> R.color.cgl25
            AppTheme.RED -> R.color.crl25
            AppTheme.NOTHING -> R.color.cnol25
        }
    }
}
